### PR TITLE
Bump to golang 1.7

### PR DESCRIPTION
**Pull request by caglar10ur**

### CODE REVIEW

**Comment by mdubya66** [comment]
I approve this change now that Go Lang has GA'd.

File: cmd/docker/main.go [hunk 1]
```diff
@@ -176,7 +176,28 @@ func startServerWithOptions(cli *CliOptions) *apiserver.Server {
 		Version: "1.22", //dockerversion.Version,
 	}
 
-	tlsConfig := tlsconfig.ServerDefault
+	copyTLSConfig := func(c *tls.Config) *tls.Config {
```
Comment (fdawg4l) [inline cmd/docker/main.go]: Does this need to be in an anonymous func?  I only see it used once immediately after declaring.

File: vendor/manifest [hunk 2]
```diff
@@ -34,7 +34,7 @@
 			"importpath": "github.com/HewlettPackard/gas",
 			"repository": "https://github.com/HewlettPackard/gas",
 			"vcs": "git",
-			"revision": "12d370b11b9e6a94368ebd229b020bb55b51fe69",
+			"revision": "37205e9afa56150a634a1d0d19347446e9b27f17",
```
Comment (fdawg4l) [inline vendor/manifest]: Is the bump for gas required for go1.7?

File: cmd/docker/main.go (hunk 1 above)
Comment (caglar10ur) [inline cmd/docker/main.go]: Could be, will do that. I added this as a temporary workaround until the library we use fixes their code :)

File: cmd/docker/main.go [hunk 3]
```diff
@@ -176,7 +176,27 @@ func startServerWithOptions(cli *CliOptions) *apiserver.Server {
 		Version: "1.22", //dockerversion.Version,
 	}
 
-	tlsConfig := tlsconfig.ServerDefault
+	tlsConfig := func(c *tls.Config) *tls.Config {
```
Comment (hickeng) [inline cmd/docker/main.go]: ?

Comment (caglar10ur) [inline cmd/docker/main.go]: go vet is/was whining about ServerDefault copying locks

Comment (caglar10ur) [inline cmd/docker/main.go]: cmd/docker/main.go:179: assignment copies lock value to tlsConfig: crypto/tls.Config contains sync.Once contains sync.Mutex

**Comment by hickeng** [comment]
LGTM - but why the need to explicitly expand the tls config?

**Comment by fdawg4l** [comment]
LGTM

**Comment by caglar10ur** [comment]
♻️

**Pull request merged**